### PR TITLE
PHOENIX-5522 If index is disabled, IndexUpgradeTool should not fail d…

### PR DESCRIPTION
**Pull request by gokceni**
…uring indexTool run

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java [hunk 1]
```diff
@@ -491,6 +491,15 @@ private int rebuildIndexes(String dataTable, IndexTool indexingTool) {
             try {
                 LOGGER.info("Rebuilding index: " + String.join(",", args));
                 if (!dryRun) {
+                    // If the index is in DISABLED state, indexTool will fail. First to ALTER REBUILD ASYNC
+                    String stmt = String.format("ALTER INDEX %s ON %s REBUILD ASYNC", indexName,
+                            SchemaUtil.getTableName(schema, baseTable));
+                    try {
+                        conn.createStatement().execute(stmt);
+                    } catch (SQLException ex) {
+                        LOGGER.warning("Something went wrong while running ALTER REBUILD ASYNC the index "
```
Comment (gjacoby126) [inline phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java]: nit: "ASYNC on the index "

File: phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java [hunk 2]
```diff
@@ -491,6 +491,15 @@ private int rebuildIndexes(String dataTable, IndexTool indexingTool) {
             try {
                 LOGGER.info("Rebuilding index: " + String.join(",", args));
                 if (!dryRun) {
+                    // If the index is in DISABLED state, indexTool will fail. First to ALTER REBUILD ASYNC
```
Comment (gjacoby126) [inline phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java]: nit: Good for the comment to make clear that ALTER INDEX REBUILD ASYNC will set the index state to BUILDING, which is safe to make ACTIVE later.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ParameterizedIndexUpgradeToolIT.java [hunk 3]
```diff
@@ -221,25 +235,25 @@ private void checkOldIndexingCoprocessors(String [] indexList, String [] tableLi
...
-            {true, false, true},
-            {false, true, true},
-            {true, true, true},
-            {false, false, false},
-            {true, false, false},
-            {false, true, false},
-            {true, true, false}
-        });
+        return Arrays.asList(new Object[][] { { false, false, true, true },
+                { false, false, true, false }, { true, false, true, true },
```
Comment (gjacoby126) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/ParameterizedIndexUpgradeToolIT.java]: how long do all of these take to run now? Are there combinations that are redundant (or orthogonal and can be tested by a single test)?

Comment (priyankporwal) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/ParameterizedIndexUpgradeToolIT.java]: +1 .. The test matrix keeps getting doubled for every variant. Perhaps mix these up with some random combination of isNamespaceEnbaled.

File: phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java (hunk 2 above)
Comment (gokceni) [reply phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java]: done

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ParameterizedIndexUpgradeToolIT.java (hunk 3 above)
Comment (gokceni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/ParameterizedIndexUpgradeToolIT.java]: What do you think @kadirozde ? Should I remove the combination and make just 1 index disabled?

Comment (kadirozde) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/ParameterizedIndexUpgradeToolIT.java]: The test already takes 20 min so we should not increase its time any further. Also, we can eliminate some of the existing combinations to reduce the test time as 20 min is already too long for this test.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ParameterizedIndexUpgradeToolIT.java [hunk 4]
```diff
@@ -111,18 +119,13 @@ public void setup () throws Exception {
...
-                || Boolean.toString(!isNamespaceEnabled).equals(serverProps
-                .get(QueryServices.IS_NAMESPACE_MAPPING_ENABLED))) {
-            tearDownMiniClusterAsync(1);
-        }
+        tearDownMiniClusterAsync(1);
+
         //setting up properties for namespace
         clientProps.put(QueryServices.IS_NAMESPACE_MAPPING_ENABLED,
-                Boolean.toString(isNamespaceEnabled));
+                Boolean.toString(true));
```
Comment (priyankporwal) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/ParameterizedIndexUpgradeToolIT.java]: Namespaces are not enabled by default, right? Should we set these to false instead?

Comment (gokceni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/ParameterizedIndexUpgradeToolIT.java]: Sure but I think it is good to test with namespace just to cover that too. We need to pick either. I thought this is a safer approach to test that scenario too.

Comment (priyankporwal) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/ParameterizedIndexUpgradeToolIT.java]: I am not sure which ones gives better coverage. If we are not sure, we can leave the isNameSpaceEnabled and set it to true/false on the rows from lines 235-238, as the last values in tuples: e.g. something like this.
            {false, false, true},
            {true, false, false},
            {false, true, false},
            {true, true, true}

**Pull request closed without merging**